### PR TITLE
Mks 46 create endpoint for the entity

### PR DESCRIPTION
**Pull request by sergiynikon**
only read operations for that moment

### CODE REVIEW

File: Serhii Nykon/NedoNet.API/Entities/UserViewEntity.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿namespace NedoNet.API.Entities
```
Comment (bvzayats) [inline Serhii Nykon/NedoNet.API/Entities/UserViewEntity.cs]: It should be created in another PR

File: Serhii Nykon/NedoNet.API/Repositories/IUsersRepository.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (bvzayats) [inline Serhii Nykon/NedoNet.API/Repositories/IUsersRepository.cs]: It should be implemented in another PR

File: Serhii Nykon/NedoNet.API/Repositories/UsersRepository.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
+﻿using System;
```
Comment (bvzayats) [inline Serhii Nykon/NedoNet.API/Repositories/UsersRepository.cs]: It should be implemented in another PR

File: Serhii Nykon/NedoNet.API/Services/IUsersService.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
```
Comment (bvzayats) [inline Serhii Nykon/NedoNet.API/Services/IUsersService.cs:1]: It should be implemented in another PR

File: Serhii Nykon/NedoNet.API/Services/UsersService.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
```
Comment (bvzayats) [inline Serhii Nykon/NedoNet.API/Services/UsersService.cs:1]: It should be implemented in another PR

File: Serhii Nykon/NedoNet.API/UOF/IUnitOfWork.cs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+﻿using System.Threading.Tasks;
```
Comment (bvzayats) [inline Serhii Nykon/NedoNet.API/UOF/IUnitOfWork.cs]: It should be implemented in another PR

File: Serhii Nykon/NedoNet.API/UOF/UnitOfWork.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
+﻿using AutoMapper;
```
Comment (bvzayats) [inline Serhii Nykon/NedoNet.API/UOF/UnitOfWork.cs]: It should be implemented in another PR

File: Serhii Nykon/NedoNet.API/Services/UsersService.cs [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+                    connection.Close();
+
+                    return OperationResult.Success();
+                } catch (Exception e) {
+                    return OperationResult.Error( e.Message );
+                }
+            }
+        }
+
+        private User CreateUser(SqlDataReader dr)
```
Comment (Magfiozy) [inline Serhii Nykon/NedoNet.API/Services/UsersService.cs]: Change name

Comment (sergiynikon) [reply Serhii Nykon/NedoNet.API/Services/UsersService.cs]: done

File: Serhii Nykon/NedoNet.API/Controllers/UsersController.cs [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+        [Route("user")]
+        public async Task<IActionResult> GetUserAsync([FromQuery] Guid id) {
+
+            var result = await _usersService.GetUserAsync( id );
+
+            if ( result.IsSuccess ) {
+                return Ok( result.Result );
+            }
+
+            return BadRequest( result.Result );
```
Comment (Magfiozy) [inline Serhii Nykon/NedoNet.API/Controllers/UsersController.cs]: Check all errors codes

File: Serhii Nykon/NedoNet.API/Controllers/UsersController.cs [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+        [HttpPut]
+        public IActionResult UpdateUser([FromQuery] Guid? id, [FromBody] UpdateUserEntity userEntity) {
```
Comment (Magfiozy) [inline Serhii Nykon/NedoNet.API/Controllers/UsersController.cs]: Guid should be not nullable

File: Serhii Nykon/NedoNet.API/Entities/CreateUserEntity.cs [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+        [Compare(nameof(Password))]
+        public string ConfirmPassword { get; set; }
+
+        [Required]
+        public string FirstName { get; set; }
+
+        [Required]
+        public string LastName { get; set; }
+
+        public string PhoneNumber { get; set; }
```
Comment (Magfiozy) [inline Serhii Nykon/NedoNet.API/Entities/CreateUserEntity.cs]: Add correct attributes

File: Serhii Nykon/NedoNet.API/Entities/UpdateUserEntity.cs [hunk 12]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using NedoNet.API.Data.Models;
+
+namespace NedoNet.API.Entities {
+    public class UpdateUserEntity {
+        public string Email { get; set; }
```
Comment (Magfiozy) [inline Serhii Nykon/NedoNet.API/Entities/UpdateUserEntity.cs]: Validation attributes

File: Serhii Nykon/NedoNet.API/MappingProfile.cs [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+{
+    public class MappingProfile : Profile {
+        public MappingProfile()
+        {
+            #region Entities -> Models
+
+            CreateMap<User, UserViewEntity>()
+                .ForMember(
+                    dest => dest.FullName,
+                    opts => opts.MapFrom( src => src.FirstName + " " + src.LastName ) );
```
Comment (Magfiozy) [inline Serhii Nykon/NedoNet.API/MappingProfile.cs]: Use interpolation

Comment (sergiynikon) [reply Serhii Nykon/NedoNet.API/MappingProfile.cs]: done

File: Serhii Nykon/NedoNet.API/Entities/UpdateUserEntity.cs (hunk 12 above)
Comment (sergiynikon) [reply Serhii Nykon/NedoNet.API/Entities/UpdateUserEntity.cs]: Done

File: Serhii Nykon/NedoNet.API/Entities/CreateUserEntity.cs (hunk 11 above)
Comment (sergiynikon) [reply Serhii Nykon/NedoNet.API/Entities/CreateUserEntity.cs]: done

File: Serhii Nykon/NedoNet.API/Controllers/UsersController.cs (hunk 10 above)
Comment (sergiynikon) [reply Serhii Nykon/NedoNet.API/Controllers/UsersController.cs]: Done

File: Serhii Nykon/NedoNet.API/Controllers/UsersController.cs (hunk 9 above)
Comment (sergiynikon) [reply Serhii Nykon/NedoNet.API/Controllers/UsersController.cs]: Done
